crypto/pkcs11key/key.go

```diff
@@ -307,8 +474,12 @@ func (ps *Key) openSession() (pkcs11.SessionHandle, error) {
 		// practice it appears to be okay to login to a token multiple times with the same
 		// credentials.
 		if err = ps.module.Login(session, pkcs11.CKU_USER, ps.pin); err != nil {
-			ps.module.CloseSession(session)
-			return session, err
+			if err == pkcs11.Error(pkcs11.CKR_USER_ALREADY_LOGGED_IN) {
```

Comment:
This relates to the comment above. I just happen to use a token that doesn't support multiple login to the same session. I'll add a comment.